### DISCUSSION TITLE
Express structure

### DISCUSSION THREAD

**Discussion by nimonian** [swe-delivery/swe-tube#15] [General]
I'm looking at #8  - it is neat how this uses Netlify functions!

Something to think about is how we would decompose (and maybe even slightly simplify) the structure in order to introduce it in a way that's easy to understand, without sacrificing too much realism.

I imagine we would probably omit the Netlify concerns throughout our teaching, and then refactor slightly for deployment when the time comes. It might help to have a discussion about what (for example) the Movie Database API folder structure would look like in week 7.

**Comment by accuser** [comment]
I've taken the view that the `api` (for the most part) is largely for modules post-bootcamp, but that's not to say that apprentices can't look inside. During bootcamp they can access it as they would any other 3rd-party API, either using `fetch`, or possibly a client library.

I want to add support to run the `api` service on Node.js directly, as well as locally via Netlify CLI, and also deployed. Only the last two are supported right now.

**Comment by accuser** [comment]
#17 should mitigate this. I have added an Express server Implementation. It reuses the existing router, so is a good example of decomposition.

**Reply by nimonian** [reply]
Fabulous! I'm looking through this now as we prepare to tackle weeks 5 - 8 of bootcamp. This work is absolute gold for guiding standards.
